tket/src/Predicates/PassGenerators.cpp

```diff
@@ -126,6 +126,169 @@ PassPtr gen_squash_pass(
+    if (gateset.find(OpType::CX) != gateset.end()) {
+      return CircPool::TK2_using_CX;
+    }
+    if (gateset.find(OpType::ZZMax) != gateset.end()) {
+      return CircPool::TK2_using_ZZMax;
+    }
+    if (gateset.find(OpType::AAMS) != gateset.end()) {
+      return CircPool::TK2_using_AAMS;
+    }
+  } else {
+    if (gateset.find(OpType::TK2) != gateset.end()) {
+      return CircPool::TK2_using_TK2_or_swap;
+    }
+    if (gateset.find(OpType::ZZPhase) != gateset.end()) {
+      return CircPool::TK2_using_ZZPhase_and_swap;
+    }
+    if (gateset.find(OpType::CX) != gateset.end()) {
+      return CircPool::TK2_using_CX_and_swap;
+    }
+    if (gateset.find(OpType::ZZMax) != gateset.end()) {
+      return CircPool::TK2_using_ZZMax_and_swap;
+    }
+  }
+  throw Unsupported("No known decomposition from TK2 to available gateset.");
+}
+
+PassPtr gen_auto_rebase_pass(const OpTypeSet& allowed_gates, bool allow_swaps) {
+  auto find_rebase = [allowed_gates, allow_swaps]() {
+    auto tk1_replacement = find_tk1_replacement(allowed_gates);
+    if (allowed_gates.find(OpType::CX) != allowed_gates.end() &&
+        allowed_gates.find(OpType::TK2) == allowed_gates.end() &&
+        !allow_swaps) {
+      return Transforms::rebase_factory(
+          allowed_gates, CircPool::CX(), tk1_replacement);
+    }
+    try {
+      return Transforms::rebase_factory_via_tk2(
+          allowed_gates, tk1_replacement,
+          find_tk2_replacement(allowed_gates, allow_swaps));
+    } catch (const Unsupported&) {
+    }
+    try {
+      return Transforms::rebase_factory(
+          allowed_gates, find_cx_replacement(allowed_gates), tk1_replacement);
+    } catch (const Unsupported&) {
+      throw Unsupported(
+          "No known decomposition from CX or TK2 to available gateset.");
+    }
+  };
+  Transform t = find_rebase();
+  PredicatePtrMap precons;
+  OpTypeSet all_types(allowed_gates);
+  all_types.insert(OpType::Measure);
+  all_types.insert(OpType::Collapse);
+  all_types.insert(OpType::Reset);
+  PredicatePtr postcon1 = std::make_shared<GateSetPredicate>(all_types);
+  PredicatePtr postcon2 = std::make_shared<MaxTwoQubitGatesPredicate>();
+  std::pair<const std::type_index, PredicatePtr> pair2 =
+      CompilationUnit::make_type_pair(postcon1);
+  PredicatePtrMap s_postcons{pair2, CompilationUnit::make_type_pair(postcon2)};
+  PredicateClassGuarantees g_postcons{{pair2.first, Guarantee::Clear}};
+  PostConditions pc = {s_postcons, g_postcons, Guarantee::Preserve};
+  // record pass config
+  nlohmann::json j;
+  j["name"] = "AutoRebase";
+  j["basis_allowed"] = allowed_gates;
+  j["allow_swaps"] = allow_swaps;
+  return std::make_shared<StandardPass>(precons, t, pc, j);
+}
+
+PassPtr gen_auto_squash_pass(const OpTypeSet& singleqs) {
+  auto find_squash = [singleqs]() {
+    if (singleqs.find(OpType::Rz) != singleqs.end() &&
+        singleqs.find(OpType::PhasedX) != singleqs.end()) {
+      return Transforms::squash_1qb_to_Rz_PhasedX(true);
```

Comment:
It's strange (but consistent with existing behaviour) that we set `always_squash_symbols` to `true` here but to `false` in line 277. I suggest making `false` here as well.